### PR TITLE
V0.5.1

### PR DESCRIPTION
**Pull request by dvdplm**
Prep for releasing v0.5.1

Based on #265

### CODE REVIEW

File: RELEASE-CHECKLIST.md
Comment (insipx) [reply RELEASE-CHECKLIST.md:16]: This is super manual and can be automated; I came across this: https://github.com/softprops/action-gh-release but it will take some playing around with to get working properly I think. But that would make releases way more streamlined

File: RELEASE-CHECKLIST.md
Comment (dvdplm) [inline RELEASE-CHECKLIST.md:17]: We need both clang and gcc?

Comment (dvdplm) [reply RELEASE-CHECKLIST.md:17]: Isn't there a "always say yes" flag to be added here?

File: RELEASE-CHECKLIST.md
Comment (dvdplm) [inline RELEASE-CHECKLIST.md:19]:
```suggestion
    1. build archive: `git clone https://github.com/paritytech/substrate-archive.git --branch release --single-branch && cd substrate-archive/bin/polkadot-archive && SKIP_WASM_BUILD=1 cargo build --release`
```

File: RELEASE-CHECKLIST.md
Comment (dvdplm) [inline RELEASE-CHECKLIST.md:18]:
```suggestion
    1. install rust from rustup.rs: `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`
    1. reload the shell: `source $HOME/.cargo/env`
```

File: RELEASE-CHECKLIST.md
Comment (dvdplm) [reply RELEASE-CHECKLIST.md:19]: The build fails for me, I get:

```
error: could not compile `polkadot-service`

Caused by:
  process didn't exit successfully: <LOOOONG COMMAND HERE> (signal: 9, SIGKILL: kill)
```

Not sure what's up with that, OOM killer?

Comment (niklasad1) [reply RELEASE-CHECKLIST.md:19]: yeah, you ran out of RAM most likely :)

File: RELEASE-CHECKLIST.md
Comment (insipx) [reply RELEASE-CHECKLIST.md:17]:
```suggestion
    1. install required dependencies `apt-get update && apt-get -y install git curl gcc clang`
```

& yeah we need clang for librocksdb

Comment (dvdplm) [reply RELEASE-CHECKLIST.md]: Right, but what do we need gcc for then?

File: RELEASE-CHECKLIST.md
Comment (dvdplm) [inline RELEASE-CHECKLIST.md:18]:
```suggestion
    1. install rust from rustup.rs: `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y`
```

File: RELEASE-CHECKLIST.md
Comment (dvdplm) [inline RELEASE-CHECKLIST.md:22]:
```suggestion
1. Signal the new release to devops for deployment
```

**Review by insipx — APPROVED** [review]

File: RELEASE-CHECKLIST.md
Comment (dvdplm) [inline RELEASE-CHECKLIST.md:22]:
```suggestion
1. Prepare a new release in the github UI
```

File: RELEASE-CHECKLIST.md
Comment (dvdplm) [inline RELEASE-CHECKLIST.md:23]: @insipx How about the source code, should I zip up this branch and upload it or should I wait until this branch is merged to `release` and tagged?

Comment (insipx) [reply RELEASE-CHECKLIST.md:23]: Once everything is tagged and this PR merged, publishing the release will automatically publish the zipped source for us

File: CHANGELOG.md
Comment (emostov) [inline CHANGELOG.md:74]:
```suggestion
## **[v0.3.1]**
```
?

File: RELEASE-CHECKLIST.md
Comment (emostov) [inline RELEASE-CHECKLIST.md:11]: Do we assume the maintainer knows what is going on here with setting up the postgres DB?

File: RELEASE-CHECKLIST.md
Comment (emostov) [inline RELEASE-CHECKLIST.md:15]: Should there be a bash script or Dockerfile for this?

File: CHANGELOG.md
Comment (niklasad1) [inline CHANGELOG.md:36]: this looks a little funny, how does one escape `)` in markdown? :)

File: RELEASE-CHECKLIST.md
Comment (emostov) [inline RELEASE-CHECKLIST.md:24]: Not against the release commit on the master branch to make sure all the release stuff is approved (and release notes etc make it in)?

**Review by niklasad1 — APPROVED** [review]

**Review by emostov — APPROVED** [review]
Overall things look good but had some comments

File: RELEASE-CHECKLIST.md
Comment (niklasad1) [reply RELEASE-CHECKLIST.md:11]: yeah, there's a wiki for that https://github.com/paritytech/substrate-archive/wiki/3.)-Configuring-Postgres-for-Archive

but maybe add a link to that or something

File: CHANGELOG.md
Comment (insipx) [reply CHANGELOG.md:36]: > this looks a little funny, how does one escape `)` in markdown links? :)

```suggestion
  - More on state-tracing [here](https://github.com/paritytech/substrate-archive/wiki/6.\)-State-Tracing-&-Balance-Reconciliation)
```
I think ^ this works

Good catch

File: RELEASE-CHECKLIST.md
Comment (insipx) [reply RELEASE-CHECKLIST.md:24]: Right, we either have to sync the  branches or keep release notes & changelog only on the `release` branch. Since releasing now consists of changing all the Cargo tags to frozen substrate versions which is different than the versions on master.

So I think we have to split this into two PR's:
- one that adds the release checklist & changelog updates
- the other that updates the `release` branch with the right substrate versions

we would have to do this on every release too I think -- IE update changelog in one PR and then sync up `release` in the next. 

Another option is just keeping the Changelog & Release checklist only on the `release` branch. If we do that, though, we should definitely put that in the `README` somewhere because it could be confusing for anyone looking for the changelog.

File: CHANGELOG.md
Comment (dvdplm) [reply CHANGELOG.md:36]: I think that link actually works, but it sure looks funny.

File: RELEASE-CHECKLIST.md
Comment (dvdplm) [reply RELEASE-CHECKLIST.md:15]: That would be good yes. Not today though. :)

File: RELEASE-CHECKLIST.md
Comment (dvdplm) [reply RELEASE-CHECKLIST.md:24]: I was thinking I'd make a new PR to `master` with the release checklist and changelog updates. 

The next release would branch off of the `release` branch and merge `master` into it, then merge back to `release` I presume?

File: RELEASE-CHECKLIST.md
Comment (insipx) [reply RELEASE-CHECKLIST.md:15]: I'm looking into automating the binary creation with GHActions rather than a Dockerfile, as to avoid manually setting up containerization for every release

File: RELEASE-CHECKLIST.md
Comment (insipx) [reply RELEASE-CHECKLIST.md:24]: Yes, that's how it would have to work I think

File: CHANGELOG.md
Comment (niklasad1) [reply CHANGELOG.md:36]: No, the link doesn't work, I ended up at https://github.com/paritytech/substrate-archive/wiki/6. ^

**Pull request merged**
